src/lib/db/feature-environment-store.ts

```diff
@@ -217,11 +217,17 @@ export class FeatureEnvironmentStore implements IFeatureEnvironmentStore {
     async connectProject(
         environment: string,
         projectId: string,
+        idempotent?: boolean, // default false to respect old behavior
```

Comment:
Other tests depend on this method fail when the project is already connected to the environment. Personally, I think that this method could be independent and the other methods depending on its failure can be rewritten (there's only one [here](https://github.com/Unleash/unleash/blob/38f351871918a98ea73dedade6a966d92b608369/src/lib/services/environment-service.ts#L80-L100))